lib/app.js

```diff
@@ -294,13 +294,20 @@ function uploadSite(settings, directoryRef, argv) {
 }
 
 function getSettings(argv) {
-  var settingsFile = path.resolve('./firebase.json');
-  var settingsJSON, settings;
-  if (!fs.existsSync(settingsFile)) {
-    console.log(chalk.red('Initialization Error') + ' - Directory not ' +
+  var curDir = process.cwd();
+  var settingsFile;
+  while(!fs.existsSync((settingsFile = (curDir + '/firebase.json')))) {
```

Comment:
While this is probably correct and concise, it's rather terse. We tend to avoid assignment in while loop definitions, and I tend to be a proponent of recursive functions in places like this as they can be easier to reason about